sgd.lua

```diff
@@ -28,7 +28,7 @@ function optim.sgd(opfunc, x, config, state)
    -- (0) get/update state
    local config = config or {}
    local state = state or config
-   local lr = config.learningRate or 1e-3
+   local lr = config.learningRate ~= nil and config.learningRate or 1e-3
```

Comment:
this doesn't actually preclude one from setting learningRate to 0.
even if config.learningRate = 0, the previous code will return it. The only cases where 1e-3 is returned is when config.learningRate == false or config.learningRate == nil